packages/forms-web-app/src/lib/find-target-value-in-json.js

```diff
@@ -0,0 +1,12 @@
+const findTargetValueInJSON = (obj, target) => {
+	return target in obj
+		? obj[target]
+		: Object.values(obj).reduce((acc, val) => {
+				if (acc !== undefined) return acc;
+				if (typeof val === 'object' && val) return findTargetValueInJSON(val, target);
+		  }, undefined);
+};
```

Comment:
There is quite a lot going on in this single statement, it would be good if we could simplify this a bit:

I have had a bit of a go at it although this is written in github rather than tested in an IDE so probably needs a few tweaks 

```js
const findTargetValueInJSON = (obj, target) => {
  if (target in obj) {
    return obj[target];
  }
  
  for (const value of Object.values(obj)) {
    if (typeof value === 'object' && value) {
      const result = findTargetValueInJSON(value, target);
      if (result !== undefined) {
        return result;
      }
    }
  }

  return undefined;
};
```

Reply:
looks like this works Mark - thanks, much clearer